File: core-extra/arazzo-parser/src/main/java/org/evomaster/arazzo/models/domain/AnyExpression.java
package org.evomaster.arazzo.models.domain;

import com.fasterxml.jackson.databind.JsonNode;

/**
 * Representing the model (Any | {expression})
 * The value to pass in the parameter.
 * The value can be a constant or a Runtime Expression to be evaluated
 * and passed to the referenced operation or workflow
 */
public abstract class AnyExpression {

    public AnyExpression() {
    }

    public static class Any extends AnyExpression {
        private final JsonNode jsonNode;

        public Any(JsonNode jsonNode) {
            this.jsonNode = jsonNode;
        }

        public JsonNode getJsonNode() {
            return jsonNode;
        }
    }

    public static class Expression extends AnyExpression {
        private final String expression;

        public Expression(String expression) {
            this.expression = expression;
        }

        public String getExpression() {
            return expression;
        }
    }
}

File: core/src/main/kotlin/org/evomaster/core/problem/rest/arazzo/models/AnyExpression.kt
package org.evomaster.core.problem.rest.arazzo.models

import com.fasterxml.jackson.databind.JsonNode

sealed class AnyExpression {
    data class Constant(val data: JsonNode) : AnyExpression()

    data class Expression(val expression: RuntimeExpression) : AnyExpression()
}